lint-staged.config.cjs

```diff
@@ -1,3 +1,7 @@
 module.exports = {
   '**/*.{js,jsx,cjs,mjs,ts,tsx,cts,mts,mdx}': ['prettier --write'],
 };
+
+module.exports = {
+  '*.json': ['npm run precommit'],
```

Comment:
This should probably be more specific to package.json

Reply:
module.exports = {
  '**/*/package.json': ['npm run precommit'],
};

this should work for nested directories too !